tests/action-systems-tests/dom-attributes-action.test.js

```diff
@@ -0,0 +1,347 @@
+                        value: true
+                    }
+                ],
+                remove: null
+            });
+
+            //Assert
+            assertEquals(element.getAttribute("hidden"), true);
+        });
+
+        it("should not fail if add = null is passed through", async () => {
+            //Arrange
+            const expectedValueRemove = undefined;
+            const element = await crs.call("dom", "create_element", {
+                tag_name: "div",
+                id: "testRemoved",
+                attributes: {
+                    hidden: true
+                }
+            });
+
+            document.children[1].appendChild(element);
+
+            //Act
+            await crs.call("dom_attributes", "perform", {
+                add: null,
+                remove: [
+                    {
+                        element: "#testRemoved",
+                        attr: "hidden"
+                    }
+                ]
+            });
+
+            //Assert
+            assertEquals(element.getAttribute("hidden"), expectedValueRemove);
+        });
+
+        it("should not fail if add = null or one of the values in the remove array is null and is passed through", async () => {
+            //Arrange
+            const expectedValueRemove = undefined;
+            const element = await crs.call("dom", "create_element", {
+                tag_name: "div",
+                id: "removingAttribute",
+                attributes: {
+                    hidden: true
+                }
+            });
+
+            document.children[1].appendChild(element);
+
+            //Act
+            await crs.call("dom_attributes", "perform", {
+                add: null,
+                remove: [
+                    {
+                        element: "#removingAttribute",
+                        attr: "hidden"
+                    },
+                    null
+                ]
+            });
+
+            //Assert
+            assertEquals(element.getAttribute("hidden"), expectedValueRemove);
+        });
+
+        it('should not fail if the element passed it does not exist - add', async () => {
+            //Arrange
+            const expectedValueAdd = undefined;
+            const element = await crs.call("dom", "create_element", {
+                tag_name: "div",
+                id: "existingElement",
+            });
+
+            document.children[1].appendChild(element);
+
+            //Act
+            await crs.call("dom_attributes", "perform", {
+                add: [
+                    {
+                        element: "#nonExistingElement",
+                        attr: "hidden",
+                        value: true
+                    }
+                ]
+            });
+
+            //Assert
+            assertEquals(element.getAttribute("hidden"), expectedValueAdd);
+        });
+
+        it('should not fail if the element passed it does not exist - remove', async () => {
```

Comment:
Should this not read: 'should not fail if the element passed in does not exist - remove'